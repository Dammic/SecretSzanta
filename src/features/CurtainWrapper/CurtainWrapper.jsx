import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

class CurtainWrapper extends PureComponent {
    render() {
        const { isHidden, customClass } = this.props
        return (
            <div className={classNames('curtain', customClass)}>
                <div className="curtain__wrapper">
                    <div className={classNames('curtain__panel curtain__panel--left', { hidden: isHidden })}>

                    </div>

                    <div className="curtain__prize">
                    </div>

                    <div className={classNames('curtain__panel curtain__panel--right', { hidden: isHidden })}>

                    </div>
                </div>
            </div>
        )
    }
}

export default CurtainWrapper
