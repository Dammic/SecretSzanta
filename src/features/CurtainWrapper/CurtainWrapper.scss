.curtain {
    overflow: hidden;
    width: 80%;
    height: 75.8%;
    position: absolute;
    display: flex;
    justify-content: center;

    .curtain__wrapper {
        overflow: hidden;
        width: 85.5%;
        height: 78%;
        align-self: flex-end;
    }

    .curtain__panel {
        width: 50%;
        height: 100%;
        float: left;
        position: relative;
        z-index: 2;
        transition: all 1s ease-out;
    }

    &.curtain__panel--fascist {
        height: 73.8%;
        .curtain__panel {
            background-color: #79040A; 
            border-width: 10px;
            border-color: transparent;
            background-clip: padding-box;
        }
    }

    &.curtain__panel--liberal {
        .curtain__panel {
            background-color: #063F56; 
        }
        .curtain__wrapper {
            height: 79%;
            width: 80.6%;
        }
    }

    .curtain__panel--left {
        transform: translateX(0);
        border-right: none;
        &.hidden {
            transform: translateX(-150%);
        }
    }

    .curtain__panel--right {
        transform: translateX(0);
        border-left: none;
        &.hidden {
            transform: translateX(150%);
        }
    }

    .curtain__prize {
        position: absolute;
        z-index: 1;
        width: 100%;
        height: 100%;
    }
}

