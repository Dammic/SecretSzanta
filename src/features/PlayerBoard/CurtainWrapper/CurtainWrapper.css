.curtain {
    overflow: hidden;
    width: 80%;
    height: 75.8%;
    position: absolute;
    display: flex;
    justify-content: center;

    &.fascist {
        height: 73.8%;
        & .curtainPanel {
            background-color: #79040A; 
        }
    }

    &.liberal {
        & .curtainPanel {
            background-color: #063F56; 
            filter: brightness(50%) sepia(1) hue-rotate(160deg) saturate(200%) brightness(330%)
        }
        & .curtainWrapper {
            height: 79%;
            width: 80.6%;
        }
    }
}

.curtainWrapper {
    overflow: hidden;
    width: 85.5%;
    height: 78%;
    display: flex;
    align-self: flex-end;
}

.curtainPanel {
    width: 50%;
    height: 100%;
    float: left;
    position: relative;
    z-index: 2;
    transition: all 2s ease-in;
    border-width: 2px;
    border-color: rgba(0,0,0,.5);
    background-clip: padding-box;
    /* background-image: url("../static/curtain.png"); */

    &.left {
        transform: translateX(0);
        border-right-style: solid;
        &.hidden {
            transform: translateX(-150%) rotateY(180deg);
        }
    }

    &.right {
        transform: translateX(0) rotateY(180deg);
        border-left-style: solid;
        &.hidden {
            transform: translateX(150%);
        }
    }
}
