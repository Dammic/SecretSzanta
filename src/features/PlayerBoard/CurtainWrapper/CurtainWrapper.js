import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import styles from './CurtainWrapper.css'

class CurtainWrapper extends PureComponent {
    static displayName = 'CurtainWrapper'
    static propTypes = {
        isHidden: PropTypes.bool,
        isFascist: PropTypes.bool,
    }

    render() {
        const { isHidden, isFascist } = this.props
        return (
            <div className={classNames(styles.curtain, { [styles.liberal]: !isFascist, [styles.fascist]: isFascist })}>
                <div className={styles.curtainWrapper}>
                    <div className={classNames(styles.curtainPanel, styles.left, { [styles.hidden]: isHidden })} />
                    <div className={classNames(styles.curtainPanel, styles.right, { [styles.hidden]: isHidden })} />
                </div>
            </div>
        )
    }
}

export default CurtainWrapper
