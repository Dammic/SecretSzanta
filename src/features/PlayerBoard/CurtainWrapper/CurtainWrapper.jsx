import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

class CurtainWrapper extends PureComponent {
    static propTypes = {
        isHidden: PropTypes.bool,
        customClass: PropTypes.string,
    }
    render() {
        const { isHidden, customClass } = this.props
        return (
            <div className={classNames('curtain', customClass)}>
                <div className="curtain__wrapper">
                    <div className={classNames('curtain__panel curtain__panel--left', { hidden: isHidden })} />
                    <div className={classNames('curtain__panel curtain__panel--right', { hidden: isHidden })} />
                </div>
            </div>
        )
    }
}

export default CurtainWrapper
