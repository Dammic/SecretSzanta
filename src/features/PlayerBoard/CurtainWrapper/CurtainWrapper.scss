.curtain {
    overflow: hidden;
    width: 80%;
    height: 75.8%;
    position: absolute;
    display: flex;
    justify-content: center;

    .curtain__wrapper {
        overflow: hidden;
        width: 85.5%;
        height: 78%;
        display: flex;
        align-self: flex-end;
    }

    .curtain__panel {
        width: 50%;
        height: 100%;
        float: left;
        position: relative;
        z-index: 2;
        transition: all 2s ease-in;
        border-width: 2px;
        border-color: rgba(0,0,0,.5);
        background-clip: padding-box;
    }

    &.curtain__panel--fascist {
        height: 73.8%;
        .curtain__panel {
            background-color: #79040A; 
            background-image: url("../static/curtain.png");
        }
    }

    &.curtain__panel--liberal {
        .curtain__panel {
            background-color: #063F56; 
            background-image: url("../static/curtain.png");
            filter: brightness(50%) sepia(1) hue-rotate(160deg) saturate(200%) brightness(330%)
        }
        .curtain__wrapper {
            height: 79%;
            width: 80.6%;
        }
    }

    .curtain__panel--left {
        transform: translateX(0);
        border-right-style: solid;
        &.hidden {
            transform: translateX(-150%) rotateY(180deg);
        }
    }

    .curtain__panel--right {
        transform: translateX(0) rotateY(180deg);
        border-left-style: solid;
        &.hidden {
            transform: translateX(150%);
        }
    }

    .curtain__prize {
        position: absolute;
        z-index: 1;
        width: 100%;
        height: 100%;
    }
}

