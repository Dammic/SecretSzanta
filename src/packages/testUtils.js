import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

const shallowRenderer = new ShallowRenderer()

export const expectMatchingSnapshot = (element) => {
    const renderedElement = renderer.create(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}

export const expectShallowMatchingSnapshot = (element) => {
    const renderedElement = shallowRenderer.render(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}
