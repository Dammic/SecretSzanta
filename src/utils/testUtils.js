import { mount } from 'enzyme'

export const expectMatchingSnapshot = (element) => {
    const renderedComponent = mount(element)
    expect(renderedComponent).toMatchSnapshot()
}
