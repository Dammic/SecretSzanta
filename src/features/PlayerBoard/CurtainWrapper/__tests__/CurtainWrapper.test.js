import React from 'react'
import { shallow } from 'enzyme'
import { expectMatchingSnapshot } from 'packages/testUtils'
import CurtainWrapper from '../CurtainWrapper'

const setupProps = (propsOverrides = {}, renderMethod = shallow) => {
    const props = {
        isHidden: false,
        isFascist: false,
        ...propsOverrides,
    }
    const component = renderMethod(<CurtainWrapper {...props} />)
    return { props, component }
}

describe('<CurtainWrapper />', () => {
    describe('liberal', () => {
        it('renders visible curtains', () => {
            const { props } = setupProps({ isHidden: false, isFascist: false })
            expectMatchingSnapshot(<CurtainWrapper {...props} />)
        })

        it('renders hidden curtains', () => {
            const { props } = setupProps({ isHidden: true, isFascist: false })
            expectMatchingSnapshot(<CurtainWrapper {...props} />)
        })
    })

    describe('fascist', () => {
        it('renders visible curtains', () => {
            const { props } = setupProps({ isHidden: false, isFascist: true })
            expectMatchingSnapshot(<CurtainWrapper {...props} />)
        })

        it('renders hidden curtains', () => {
            const { props } = setupProps({ isHidden: true, isFascist: true })
            expectMatchingSnapshot(<CurtainWrapper {...props} />)
        })
    })
})
