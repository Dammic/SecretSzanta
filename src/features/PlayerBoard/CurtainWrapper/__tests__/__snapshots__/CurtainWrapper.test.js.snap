// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurtainWrapper /> fascist renders hidden curtains 1`] = `
<div
  className="curtain fascist"
>
  <div
    className="curtainWrapper"
  >
    <div
      className="curtainPanel left hidden"
    />
    <div
      className="curtainPanel right hidden"
    />
  </div>
</div>
`;

exports[`<CurtainWrapper /> fascist renders visible curtains 1`] = `
<div
  className="curtain fascist"
>
  <div
    className="curtainWrapper"
  >
    <div
      className="curtainPanel left"
    />
    <div
      className="curtainPanel right"
    />
  </div>
</div>
`;

exports[`<CurtainWrapper /> liberal renders hidden curtains 1`] = `
<div
  className="curtain liberal"
>
  <div
    className="curtainWrapper"
  >
    <div
      className="curtainPanel left hidden"
    />
    <div
      className="curtainPanel right hidden"
    />
  </div>
</div>
`;

exports[`<CurtainWrapper /> liberal renders visible curtains 1`] = `
<div
  className="curtain liberal"
>
  <div
    className="curtainWrapper"
  >
    <div
      className="curtainPanel left"
    />
    <div
      className="curtainPanel right"
    />
  </div>
</div>
`;
